src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferAsSpanOverSubstring.Fixer.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+using RequiredSymbols = Microsoft.NetCore.Analyzers.Runtime.PreferAsSpanOverSubstring.RequiredSymbols;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    public abstract class PreferAsSpanOverSubstringFixer : CodeFixProvider
+    {
+        private const string SubstringStartIndexArgumentName = "startIndex";
+        private const string AsSpanStartArgumentName = "start";
+
+        private protected abstract void ReplaceNonConditionalInvocationMethodName(SyntaxEditor editor, SyntaxNode memberInvocation, string newName);
+
+        private protected abstract void ReplaceNamedArgumentName(SyntaxEditor editor, SyntaxNode invocation, string oldArgumentName, string newArgumentName);
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(PreferAsSpanOverSubstring.RuleId);
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var token = context.CancellationToken;
+            SyntaxNode root = await document.GetSyntaxRootAsync(token).ConfigureAwait(false);
+            SemanticModel model = await document.GetSemanticModelAsync(token).ConfigureAwait(false);
+            var compilation = model.Compilation;
+
+            if (root.FindNode(context.Span, getInnermostNodeForTie: true) is not SyntaxNode reportedNode || model.GetOperation(reportedNode, token) is not IInvocationOperation reportedInvocation)
+                return;
+            if (!RequiredSymbols.TryGetSymbols(compilation, out RequiredSymbols symbols))
+                return;
+
+            var bestCandidates = PreferAsSpanOverSubstring.GetBestSpanBasedOverloads(symbols, reportedInvocation, context.CancellationToken);
+            //  We only apply the fix if there is an unambiguous best overload.
```

Comment:
Nit: blank line before comment
```suggestion

            //  We only apply the fix if there is an unambiguous best overload.
```